numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -209,14 +210,48 @@ _strided_to_strided_truncate_copy(char *dst, npy_intp dst_stride,
     }
 }
 
+/*
+ * Does a strided to strided zero-padded or truncated copy for the case where
+ * unicode swapping is needed.
+ */
+static void
+_strided_to_strided_unicode_copyswap(char *dst, npy_intp dst_stride,
+                        char *src, npy_intp src_stride,
+                        npy_intp N, npy_intp src_itemsize,
+                        NpyAuxData *data)
+{
+    _strided_zero_pad_data *d = (_strided_zero_pad_data *)data;
+    npy_intp dst_itemsize = d->dst_itemsize;
+    npy_intp zero_size = dst_itemsize-src_itemsize;
+    char *_dst;
+    npy_intp characters = dst_itemsize / 4;
+    int i;
+
+    while (N > 0) {
+        memcpy(dst, src, dst_itemsize);
```

Comment:
Oh yeah, good point. I guess the tests just don't care that much about a bit reading too far ;). (I hope valgrind would show it but OK).

Just to note, I did not check the python 3 failure here, and I noticed there is a bug that casting from `S -> nonnative U` actually seems to reverse the string, and I am not sure why.